packages/client/hmi-client/src/services/models/simulation-service.ts

```diff
@@ -250,12 +242,48 @@ export async function simulationPollAction(
 	emitFn: (event: 'append-output-port' | 'update-state', ...args: any[]) => void
 ) {
 	const requestList: Promise<Simulation | null>[] = [];
+
 	simulationIds.forEach((id) => {
 		requestList.push(getSimulation(id));
 	});
 	const response = await Promise.all(requestList);
-	if (response.every((simulation) => simulation!.status === ProgressState.COMPLETE)) {
+
+	const completedSimulationIds = response
+		.filter((simulation) => simulation?.status === ProgressState.COMPLETE)
+		.map((simulation) => simulation!.id);
```

Comment:
I think this is a typo? `simulation!.id` instead of `simulation?.id` or is this `!.` a new thing?

Reply:
Not a typo here -` !.` is the Non-null assertion operator in typescript: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator.

In this context it asserts the simulation variable is non-null and non-undefined